### PR TITLE
Release/4.0.4

### PR DESCRIPTION
**Pull request by iakov-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cfn-templates/cid-cfn.yml [hunk 1]
```diff
@@ -655,17 +655,21 @@ Resources:
...
-                      print('status=', crawler_status)
-                      if crawler_status in ('READY', 'STOPPING'):
-                          print("Stop waiting")
-                          break
-                  else:
-                      return (True, f"Timeout exceeded. Crawler '{CRAWLER}' did not complete. This is not a fatal error and the rest of the deployment will continue.")
+                  try:
+                      start_time = time.time()
+                      while time.time() - start_time < timeout_seconds:
+                          time.sleep(10)
```
Comment (vmindru) [inline cfn-templates/cid-cfn.yml:661]: why 10 s?

File: cfn-templates/cid-cfn.yml [hunk 2]
```diff
@@ -655,17 +655,21 @@ Resources:
...
+                          crawler_status = glue.get_crawler(Name=CRAWLER)['Crawler']['State']
+                          print('status=', crawler_status)
```
Comment (vmindru) [inline cfn-templates/cid-cfn.yml:663]: print(f"status={crawler_status}" ) is more beautifull :)

File: cfn-templates/cid-cfn.yml [hunk 3]
```diff
@@ -655,17 +655,21 @@ Resources:
...
+                          if crawler_status in ('READY', 'STOPPING'):
+                              print("Stop waiting")
+                              break
+                      else:
+                          return (True, f"Timeout exceeded. Crawler '{CRAWLER}' did not complete. This is not a fatal error and the rest of the deployment will continue.")
```
Comment (vmindru) [inline cfn-templates/cid-cfn.yml:668]: maybe just throw an exception here, a non intertuptive one?

**Review by vmindru — APPROVED** [review]

File: cfn-templates/cid-cfn.yml (hunk 2 above)
Comment (iakov-aws) [inline cfn-templates/cid-cfn.yml:663]:
```suggestion
                          print(f'status = {crawler_status}')
```

File: cfn-templates/cid-cfn.yml (hunk 1 above)
Comment (iakov-aws) [reply cfn-templates/cid-cfn.yml:661]: average crawler runs about 60 s 

we can change the wait to 1 not an issue

```suggestion
                          time.sleep(1)
```

File: cfn-templates/cid-cfn.yml (hunk 3 above)
Comment (iakov-aws) [reply cfn-templates/cid-cfn.yml:668]: we do not have non intertuptive exceptions here

**Pull request merged**